### PR TITLE
Add an Opus encode path to the source role

### PR DESCRIPTION
**Pull request by chrisuthe**
Part 5/6 of the source@v1 stack (tracker: #95).

**What it adds:** an Opus encode path for the source role via micro-opus (already a dependency — its encoder is compiled in): one RFC 6716 packet per chunk, encoder lookahead reflected in the chunk timestamps, fail-closed validation of opus-legal rates/frame durations, and codec linking gated on player OR source.

**How it's used:** set `codec = OPUS` (plus bitrate/complexity) in `SourceRoleConfig`; everything else is unchanged. Verified by encode→decode round-trip tests against the library's own decoder.

**Validation:** the Opus path ran end-to-end against Music Assistant's `sendspin_source` ingest on a dev MA server (same interop shims as part 3): one RFC 6716 packet per chunk decoded by the server, timestamp continuity within tens of microseconds across the stream.

### CODE REVIEW

File: src/source_encoder_opus.cpp [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        return 0;
+    }
+
+    // `in` sits behind the 9-byte wire header and is not int16-aligned, so copy to the aligned
+    // scratch; encoding into the packet scratch (never `out`) is what honors in == out
+    memcpy(this->pcm_scratch_.data(), in, in_len);
+    const opus_int32 written =
+        opus_encode(this->encoder_state_.as<OpusEncoder>(), this->pcm_scratch_.as<opus_int16>(),
+                    static_cast<int>(in_len / this->bytes_per_frame_), this->packet_scratch_.data(),
+                    static_cast<opus_int32>(MAX_PACKET_BYTES));
```
[the remark this answers is not in the document]
Comment (chrisuthe) [reply src/source_encoder_opus.cpp]: Fixed in e853ac7: the packet encodes straight into the payload area (the aligned PCM copy already satisfies the aliasing contract), dropping the 4 KB scratch and the per-chunk copy. One deliberate semantic: libopus treats the capacity as a hard packet cap and degrades to fit rather than erroring, which the capacity test now pins; the task always offers MAX_PACKET_BYTES, so in-tree streams never degrade.

File: src/source_role.cpp [hunk 2]
```diff
@@ -32,34 +35,84 @@ namespace sendspin {
...
-        // deliberately enforced as hard as the MUST
-        SS_LOGE(TAG, "Rejecting source config: chunk_duration_ms %u outside [%u, %u]",
-                config.chunk_duration_ms, SourceRoleConfig::CHUNK_MIN_MS,
-                SourceRoleConfig::CHUNK_MAX_MS);
-        valid = false;
+    if (config.codec == SendspinCodecFormat::OPUS) {
+        // libopus accepts exactly these rates; a 44100 line-in must use PCM or resample
+        static constexpr uint32_t OPUS_SAMPLE_RATES[] = {8000, 12000, 16000, 24000, 48000};
+        // Single legal Opus frames within the spec's chunk bounds
+        static constexpr uint32_t OPUS_CHUNK_DURATIONS_MS[] = {10, 20, 40, 60};
```
[the remark this answers is not in the document]
Comment (chrisuthe) [reply src/source_role.cpp]: Fixed in e853ac7: 5 ms joins the accepted Opus durations (legal frame at the spec minimum), with the error text, config docs, and validation test flipped to an accepting boundary case.

File: include/sendspin/config.h [hunk 3]
```diff
@@ -299,14 +300,32 @@ struct SourceRoleConfig {
...
-    uint32_t sample_rate{DEFAULT_SOURCE_SAMPLE_RATE};  ///< Capture sample rate in Hz; must be > 0
+    /// @brief Capture sample rate in Hz; must be > 0. OPUS accepts only libopus's rates:
+    /// 8000, 12000, 16000, 24000, or 48000 (a 44100 line-in must use PCM or resample upstream)
+    uint32_t sample_rate{DEFAULT_SOURCE_SAMPLE_RATE};
 
     /// @brief Outbound chunk duration in milliseconds, validated against the spec bounds
-    /// [CHUNK_MIN_MS, CHUNK_MAX_MS]
+    /// [CHUNK_MIN_MS, CHUNK_MAX_MS]. OPUS accepts only 10, 20, 40, or 60 (one
+    /// chunk is exactly one legal Opus frame), so the PCM default of 25 is rejected for OPUS
     uint32_t chunk_duration_ms{DEFAULT_CHUNK_MS};
```
[the remark this answers is not in the document]
Comment (chrisuthe) [reply include/sendspin/config.h:331]: Fixed in 0b51d56 (part 3, where the default lives): the default chunk duration is now 20 ms — a legal Opus frame — so switching codec alone keeps a valid config; the config reference documents the property.

File: include/sendspin/config.h [hunk 4]
```diff
@@ -273,9 +273,10 @@ struct VisualizerRoleConfig {
 /// @brief Configuration for the source role (audio capture streamed to the server)
 ///
 /// The configured format is the contract for every stream the role opens: there is no
-/// negotiation, and write_audio() bytes are forwarded untouched. An invalid config leaves the
-/// role added but inert (logged at ERROR; the role is not advertised and never streams) --
-/// spec-invalid values are rejected, never clamped or repaired.
+/// negotiation, and write_audio() consumes PCM in exactly that format (sent untouched for the
+/// PCM codec, encoded chunk-by-chunk for OPUS). An invalid config leaves the role added but
+/// inert (logged at ERROR; the role is not advertised and never streams) -- spec-invalid values
+/// are rejected, never clamped or repaired.
```
[the remark this answers is not in the document]
Comment (chrisuthe) [reply include/sendspin/config.h:279]: Fixed in e853ac7: write_audio's header doc scopes the untouched-bytes promise to PCM and names the Opus encode behavior; the integration guide's config table (part 6) already documents the codec and the bitrate/complexity fields.

File: src/source_task.cpp [hunk 5]
```diff
@@ -33,9 +34,9 @@ namespace sendspin {
 static const char* const TAG = "sendspin.source_task";
 
 /// @brief Same budget as the sync task. Host -O2 -fstack-usage measures the deepest task-path
-/// chain (stream -> send_chunk -> event wait) near 0.6 KB; the budget's remainder is headroom
-/// for the ESP transport send path and platform differences, pending an on-target high-water
-/// measurement
+/// chain (stream -> send_chunk -> event wait) near 0.6 KB; the remainder is headroom for the
+/// ESP transport send path pending an on-target high-water measurement. Opus working buffers
+/// live on micro-opus's per-thread pseudostack, not here.
 static constexpr size_t SOURCE_TASK_STACK_SIZE = 6192;
```
[the remark this answers is not in the document]
Comment (chrisuthe) [reply src/source_task.cpp:41]: Fixed in c8a939a: Kconfig now refuses the source role under OPUS_USE_ALLOCA (the task stack budget assumes the per-thread pseudostack) and under the non-thread-safe pseudostack when the player role is also enabled (one shared arena, two concurrent codec threads), with the assumption named at the stack constant. Note the player's sync task has the same alloca exposure on main independently of this stack — worth its own follow-up guard.

File: src/source_task.cpp [hunk 6]
```diff
@@ -111,16 +112,26 @@ bool SourceTask::init(SourceRole::Impl* source_impl, SendspinClient* client,
         return false;
     }
 
-    if (!this->staging_.allocate(SOURCE_WIRE_HEADER_SIZE + this->chunk_bytes_,
+    size_t payload_capacity = this->chunk_bytes_;
+    if (config.codec == SendspinCodecFormat::OPUS) {
+        auto opus_encoder = std::make_unique<OpusSourceEncoder>();
+        if (!opus_encoder->init(config)) {
+            return false;  // Cause already logged; no streaming without a working encoder
+        }
```
[the remark this answers is not in the document]
Comment (chrisuthe) [reply src/source_task.cpp:124]: Fixed in b97b0fd (part 3, where init() lives): the event flags are created last, after every fallible allocation, so is_initialized() now means fully initialized and a retried start cannot accept a half-built task.

**Comment by chrisuthe** [comment]
On the suppressed items: the three stale references to the old 25 ms default (validation comment, config reference, opus test baseline) are fixed in c8a939a. The two dependency-documentation claims are already correct in part 6 (#117), where the stack's docs land: the integration guide's build-options section and internals both state micro-flac is player-only while micro-opus is gated on player or source.
